File: sahil_sable/frontend/lib/models/book.dart
class Book {
  final String id;
  final String title;
  final String author;
  final String genre;
  final double price;
  final int publishedYear;
  final String? createdDate;
  final String? imageUrl;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.genre,
    required this.price,
    required this.publishedYear,
    this.createdDate,
    this.imageUrl,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['_id'],
      title: json['title'],
      author: json['author'],
      genre: json['genre'],
      price: (json['price'] as num).toDouble(),
      publishedYear: json['publishedYear'],
      createdDate: json['createdDate'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'author': author,
      'genre': genre,
      'price': price,
      'publishedYear': publishedYear,
      'imageUrl': imageUrl,
    };
  }
}

File: sahil_sable_10/backend/models/Book.js
import mongoose from 'mongoose';

const bookSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true
  },
  author: {
    type: String,
    required: true,
    trim: true
  },
  genre: {
    type: String,
    required: true,
    trim: true
  },
  price: {
    type: Number,
    required: true,
    min: 0
  },
  publishedYear: {
    type: Number,
    required: true
  },
  imageUrl: {
    type: String, // Store path to the image
    required: false
  },
  createdDate: {
    type: Date,
    default: Date.now
  }
});

const Book = mongoose.model('Book', bookSchema);

export default Book;
